### PR TITLE
Feat: refactor resources/data to use the new serialization/deserializ…

### PR DESCRIPTION
**Pull request by TomerHeber**
…ation logic

### Solution

part of #376 

Refactored project resource and project data resource.

### CODE REVIEW

File: env0/resource_project.go [hunk 1]
```diff
@@ -82,10 +70,12 @@ func resourceProjectRead(ctx context.Context, d *schema.ResourceData, meta inter
 
 	project, err := apiClient.Project(d.Id())
 	if err != nil {
-		return diag.Errorf("could not get project: %v", err)
+		return ResourceGetFailure("project", d, err)
```
Comment (TomerHeber) [inline env0/resource_project.go:73]: this should handle drift issues as well.

File: env0/resource_project.go [hunk 2]
```diff
@@ -94,18 +84,16 @@ func resourceProjectUpdate(ctx context.Context, d *schema.ResourceData, meta int
 	apiClient := meta.(client.ApiClientInterface)
 
 	id := d.Id()
-	payload := client.ProjectCreatePayload{
-		Name:        d.Get("name").(string),
-		Description: d.Get("description").(string),
+	var payload client.ProjectCreatePayload
+
+	if err := readResourceData(&payload, d); err != nil {
```
Comment (GiliFaroEnv0) [inline env0/resource_project.go:89]: so from now on we first read the resource and deserialized it
and before write we serialize it?

**Review by GiliFaroEnv0 — COMMENTED** [review]
@TomerHeber approve just a Q to confirm I understand correctly

File: env0/resource_project.go (hunk 2 above)
Comment (TomerHeber) [reply env0/resource_project.go:89]: @GiliFaroEnv0 

readResourceData extracts from resourceData and write to the interface (E.g. `var payload client.ProjectCreatePayload` in this case).

writeResourceData extracts from the interface (a struct usually) and write to the resourceData).

Note:
For most cases this works well. But in some cases manual d.Get("...") or d.Set("...") is required.
These two functions use golang reflection.

**Review by GiliFaroEnv0 — APPROVED** [review]
Great job 👍

**Pull request merged**